asmcomp/cmm.mli

```diff
@@ -134,10 +134,10 @@ and expression =
   | Clet of Ident.t * expression * expression
   | Cassign of Ident.t * expression
   | Ctuple of expression list
-  | Cop of operation * expression list
+  | Cop of operation * expression list * Debuginfo.t
   | Csequence of expression * expression
   | Cifthenelse of expression * expression * expression
-  | Cswitch of expression * int array * expression array
+  | Cswitch of expression * int array * expression array * Debuginfo.t
```

Comment:
The idea was to put debuginfo on those constructs which correspond to actual instructions in the output. We decided that mostly meant `Cop`, `Cswitch` and `Cifthenelse`, but then I think Mark decided to skip `Cifthenelse` since it didn't turn out to be particularly important.

I'm not too sure about putting debuginfo on everything. Things like `let` do not really correspond to resulting instructions, or indeed to a specific piece of the source program, so it is not clear that attaching debuginfo to it really makes sense.